packages/background/src/index.integration.test.js

```diff
@@ -2365,6 +2401,137 @@ describe('integration test', function () {
         const [, n2] = db.getNetwork()
         expect(n2.name).toBe(ETH_MAINNET_NAME)
         expect(n2.endpoint).toBe(ETH_LOCALNET_RPC_ENDPOINT)
+        await request({
+          method: 'wallet_deleteNetwork',
+          params: {
+            password,
+            networkId: db.getNetworkByName(ETH_MAINNET_NAME)[0].eid,
+          },
+        })
+        res = request({
+          _popup: true,
+          method: 'wallet_addEthereumChain',
+          params: [
+            {
+              chainId: ETH_LOCALNET_CHAINID,
+              chainName: ETH_MAINNET_NAME,
+              nativeCurrency: {
+                name: 'ETH',
+                symbol: 'ETH',
+                decimals: DEFAULT_CURRENCY_DECIMALS,
+              },
+              rpcUrls: [ETH_LOCALNET_RPC_ENDPOINT],
+            },
+          ],
+        })
+        expect((await res).result).toBe('__null__')
+      })
+
+      test('error call adding duplicate endpoint network', async () => {
+        await request({method: 'wallet_generatePrivateKey'}).then(({result}) =>
+          request({
+            method: 'wallet_importPrivateKey',
+            params: {privateKey: result, password},
+          }),
+        )
+
+        const [a1] = db.getAccount()
+        res = request({
+          method: 'cfx_requestAccounts',
+          _inpage: true,
+          _origin: 'foo.site',
+          networkName: CFX_MAINNET_NAME,
+        })
+
+        await waitForExpect(() => expect(db.getAuthReq().length).toBe(1))
+
+        await request({
+          method: 'wallet_requestPermissions',
+          params: {
+            permissions: [{cfx_accounts: {}}],
+            accounts: [a1.eid],
+            authReqId: db.getAuthReq()[0].eid,
+          },
+          networkName: CFX_MAINNET_NAME,
+          _popup: true,
+        })
+
```

Comment:
don't rely on logic that cross `test` boundaries
db is cleaned up on the start of every `test`